crates/ra_hir_ty/src/tests.rs

```diff
@@ -37,6 +37,14 @@ use crate::{
 // against snapshots of the expected results using insta. Use cargo-insta to
 // update the snapshots.
 
+fn with_tracing_logs<T>(action: impl FnOnce() -> T) -> T {
```

Comment:
An alternative would be guard based api: let _tracing = with_tracing_logs. It would be a tad more rustic, and will also avoid impl vs dyn question.

Reply:
Turns out tracing does have such an API!